laborator/lab-09/README.md

```diff
@@ -0,0 +1,216 @@
+# Laborator 09: Interactiunea C-assembly
+Având în vedere că limbajul de asamblare prezintă dificultăți atât în citirea cât și în dezvoltarea codului, tendința generală este aceea de a se migra către limbaje de nivel înalt (care sunt mult mai ușor de citit și oferă un API mult mai ușor de utilizat). Cu toate acestea, tot există situații în care, din rațiuni de optimizare, se folosesc mici rutine assembly care sunt integrate în modulul limbajului de nivel inalt.
+
+În acest laborator vom vedea cum se pot integra module de assembly în programe C și viceversa.
+
+## Utilizarea procedurilor assembly în funcții C
+Pentru ca un program C să ajungă să fie executat, este necesar ca acesta să fie tradus în codul mașina al procesorului; aceasta este sarcina unui compilator. Având în vedere că acest cod rezultat în urma compilării nu este întotdeauna optim, în anumite cazuri se preferă înlocuirea unor porțiuni de cod scris în C cu porțiuni de cod assembly care să facă același lucru, însă cu o performanță mai bună.
+
+### Declararea procedurii
+Pentru a ne asigura că procedura assembly și modulul C se vor combina cum trebuie și vor fi compatibile, următorii pași trebuie urmați:
+- declararea labelului procedurii ca fiind global, folosing directiva GLOBAL. Pe lângă asta, orice date care vor fi folosite de către procedură trebuie declarate ca fiind globale.
+- folosirea directivei EXTERN pentru a declara procedurile și datele globale ca fiind externe.
+
+### Setarea stivei
+Atunci când se intră intr-o procedură, este necesar să se seteze un stack frame către care să se trimită parametrii. Desigur, dacă procedura nu primește parametri, acest pas nu este necesar. Așadar, pentru a seta stiva, trebuie inclus următorul cod: 
+```bash
```

Comment:
Textul de după backtick-uri este numele pentru syntax highlighting. Github folosește linguist pentru syntax highlight, conform [documentației](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks). Pentru assembly, ar trebui să merge `Assembly` conform fișierului [yaml](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L375) (este menționat la finalul fișierului de documentație. Pentru C este `C`.